include/bouffalolab/bl60x/bflb_soc.h

```diff
@@ -0,0 +1,234 @@
+#define TZC_NSEC_BASE    ((uint32_t)0x40006000) /*!< Trustzone control none-security base address */
+#define EF_DATA_BASE     ((uint32_t)0x40007000)
+#define EF_CTRL_BASE     ((uint32_t)0x40007000)
+#define CCI_BASE         ((uint32_t)0x40008000)
+#define L1C_BASE         ((uint32_t)0x40009000) /*!< L1 cache config base address */
+#define UART0_BASE       ((uint32_t)0x4000A000)
+#define UART1_BASE       ((uint32_t)0x4000A100)
+#define SPI_BASE         ((uint32_t)0x4000A200)
+#define I2C_BASE         ((uint32_t)0x4000A300)
+#define PWM_BASE         ((uint32_t)0x4000A400)
+#define TIMER_BASE       ((uint32_t)0x4000A500)
+#define IR_BASE          ((uint32_t)0x4000A600)
+#define SF_CTRL_BASE     ((uint32_t)0x4000B000)
+#define SF_CTRL_BUF_BASE ((uint32_t)0x4000B700)
+#define DMA_BASE         ((uint32_t)0x4000C000)
+#define SDU_BASE         ((uint32_t)0x4000D000)
+#define PDS_BASE         ((uint32_t)0x4000E000) /*!< Power down sleep module base address */
+#define HBN_BASE         ((uint32_t)0x4000F000) /*!< Hibernate module base address */
+#define AON_BASE         ((uint32_t)0x4000F000) /*!< Always on module base address */
+#define HBN_RAM_BASE     ((uint32_t)0x40010000)
+
+typedef enum {
+    BL_AHB_SLAVE1_GLB = 0x00,
+    BL_AHB_SLAVE1_RF = 0x01,
+    BL_AHB_SLAVE1_GPIP_PHY_AGC = 0x02,
+    BL_AHB_SLAVE1_SEC_DBG = 0x03,
+    BL_AHB_SLAVE1_SEC = 0x04,
+    BL_AHB_SLAVE1_TZ1 = 0x05,
+    BL_AHB_SLAVE1_TZ2 = 0x06,
+    BL_AHB_SLAVE1_EFUSE = 0x07,
+    BL_AHB_SLAVE1_CCI = 0x08,
+    BL_AHB_SLAVE1_L1C = 0x09,
+    BL_AHB_SLAVE1_RSVD0A = 0x0A,
+    BL_AHB_SLAVE1_SFC = 0x0B,
+    BL_AHB_SLAVE1_DMA = 0x0C,
+    BL_AHB_SLAVE1_SDU = 0x0D,
+    BL_AHB_SLAVE1_PDS_HBN_AON_HBNRAM = 0x0E,
+    BL_AHB_SLAVE1_RSVD0F = 0x0F,
+    BL_AHB_SLAVE1_UART0 = 0x10,
+    BL_AHB_SLAVE1_UART1 = 0x11,
+    BL_AHB_SLAVE1_SPI = 0x12,
+    BL_AHB_SLAVE1_I2C = 0x13,
+    BL_AHB_SLAVE1_PWM = 0x14,
+    BL_AHB_SLAVE1_TMR = 0x15,
+    BL_AHB_SLAVE1_IRR = 0x16,
+    BL_AHB_SLAVE1_CKS = 0x17,
+    BL_AHB_SLAVE1_MAX = 0x18,
+} BL_AHB_Slave1_Type;
+
+typedef enum {
+    BL_AHB_SEC_ENG_AES0 = 0,
+    BL_AHB_SEC_ENG_AES1,
+    BL_AHB_SEC_ENG_SHA0,
+    BL_AHB_SEC_ENG_SHA1,
+} BL_AHB_Sec_Eng_Type;
+
+typedef enum {
+    BL_AHB_DMA0_CH0 = 0,
+    BL_AHB_DMA0_CH1,
+    BL_AHB_DMA0_CH2,
+    BL_AHB_DMA0_CH3,
+    BL_AHB_DMA0_CH4,
+} BL_AHB_DMA0_CHNL_Type;
+
+typedef enum {
+    BL_CORE_MASTER_IBUS_CPU = 0,
+    BL_CORE_MASTER_DBUS_CPU,
+    BL_CORE_MASTER_BUS_S2F,
+    BL_CORE_MASTER_MAX,
+} BL_Core_Master_Type;
+
+typedef enum {
+    BL_CORE_SLAVE0_DTCM_CPU = 0,
+    BL_CORE_SLAVE0_MAX,
+} BL_Core_Slave0_Type;
+
+typedef enum {
+    BL_CORE_SLAVE1_XIP_CPU = 0,
+    BL_CORE_SLAVE1_ITCM_CPU,
+    BL_CORE_SLAVE1_ROM,
+    BL_CORE_SLAVE1_MAX,
+} BL_Core_Slave1_Type;
+
+typedef enum {
+    BL_CORE_SLAVE2_F2S = 0,
+    BL_CORE_SLAVE2_MAX,
+} BL_Core_Slave2_Type;
+
+/**
+  * @}
+  */
+#include <stdint.h>
```

Comment:
Same here, please drop all

Reply:
nop

In C you don't need add any #include in header files. The ideal is always track where you really need the dependencies, which will be in the *.c files. This way we complete isolate all the dependencies.

Reply:
Drop all and include in the *.c file

./bl60x/sf_ctrl_reg.h:39:#include <stdint.h>
./bl60x/tzc_sec_reg.h:39:#include <stdint.h>
./bl60x/extra_defines.h:7:#include <stdint.h>
./bl60x/ef_data_reg.h:39:#include <stdint.h>
./bl60x/pds_reg.h:39:#include <stdint.h>
./bl60x/glb_reg.h:39:#include <stdint.h>
./bl60x/aon_reg.h:39:#include <stdint.h>
./bl60x/l1c_reg.h:39:#include <stdint.h>
./bl60x/hbn_reg.h:39:#include <stdint.h>
./bl60x/bflb_soc.h:232:#include <stdint.h>
./bl60x/ef_ctrl_reg.h:39:#include <stdint.h>

Reply:
Done